### PR TITLE
lua, bump version

### PR DESCRIPTION
**Pull request by Begasus**

### CODE REVIEW

**Comment by Begasus** [comment]
Recipe should be OK, checked on gcc2h with x86_gcc2 and _x86
I see other systems using libreadline to link to also, maybe it can be added? (works also)

**Comment by Begasus** [comment]
Maybe we could rename this to lua54 for testing purpose?

File: dev-lang/lua/lua-5.4.0.recipe [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+	PATCHES+="
+		lua-$portVersion.gcc2.patchset
+	"
+fi
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+SECONDARY_ARCHITECTURES="x86_gcc2 x86"
+
+PROVIDES="
+	lua$secondaryArchSuffix = $portVersion compat >= 5.4
```
Comment (Begasus) [inline dev-lang/lua/lua-5.4.0.recipe]: Do we need a compat here?

File: dev-lang/lua/lua-5.4.0.recipe [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	cmd:lua$secondaryArchSuffix = $portVersion compat >= 5.4
+	cmd:lua5.4$secondaryArchSuffix = $portVersion compat >= 5.4
+	cmd:luac$secondaryArchSuffix = $portVersion compat >= 5.4
+	cmd:luac5.4$secondaryArchSuffix = $portVersion compat >= 5.4
+	lib:liblua$secondaryArchSuffix = $portVersion compat >= 5.4
```
Comment (Begasus) [inline dev-lang/lua/lua-5.4.0.recipe]: We could start using libVersionCompat here, but would it be worth the effort?

**Comment by Begasus** [comment]
OK, moved the recipe name to lua54-5.4.0~beta.recipe (and the patches, still have the previous ones around) and added variables that should make bumping 5.4.* version easier (IMHO)
Some variables will need changes once stable is released, at least this wont interfere with our default 5.3.5 version ...
Reviews welcome :)

File: dev-lang/lua/lua54-5.4.0~beta.recipe [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+REQUIRES_devel="
+	lua54$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:make
+	cmd:gcc$secondaryArchSuffix
```
Comment (korli) [inline dev-lang/lua/lua54-5.4.0~beta.recipe]: cmd:make comes after cmd:gcc

Comment (Begasus) [reply dev-lang/lua/lua54-5.4.0~beta.recipe]: Fixed (this should be done for lua 5.3.5 also) :) (leftover from there)

File: dev-lang/lua/lua54-5.4.0~beta.recipe [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	"
+
+PROVIDES_devel="
+	lua54${secondaryArchSuffix}_devel = $portVersion
+	devel:liblua$secondaryArchSuffix = $libVersionCompat
+	"
+REQUIRES_devel="
+	lua54$secondaryArchSuffix == $portVersion base
+	"
+
```
Comment (Begasus) [inline dev-lang/lua/lua54-5.4.0~beta.recipe]: Shouldn't we add a CONFLICTS here (like in lua52)?

Comment (korli) [reply dev-lang/lua/lua54-5.4.0~beta.recipe]: Probably, if the link is named the same.

Comment (Begasus) [reply dev-lang/lua/lua54-5.4.0~beta.recipe]: How do I check? lua.pc is called the same in all versions?

Comment (Begasus) [reply dev-lang/lua/lua54-5.4.0~beta.recipe]: Well, lua$libVersion.pc is version related ... (but writes over the lua.pc file if you check in Terminal)

Comment (Begasus) [reply dev-lang/lua/lua54-5.4.0~beta.recipe]: added CONFLICTS

**Comment by Begasus** [comment]
I think this should be OK to merge, comments welcome :)

**Comment by Begasus** [comment]
@korli any ideas on this one? Or wait until the official release?

**Pull request merged**